### PR TITLE
track the log string buffer size 

### PR DESCRIPTION
**Pull request by weijjia**
#145: Track the log string buffer size for sample app, if it exceeds the max limit, will re-assign the string buffer to avoid the OOM.

### CODE REVIEW

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java [hunk 1]
```diff
@@ -33,20 +33,26 @@
...
+    private StringBuilder mLogs;
+    private int mLogSize;
 
     @Override
     public void onCreate() {
         super.onCreate();
-        mLogs = new StringBuffer();
-
+        mLogs = new StringBuilder();
+
```
Comment (iambmelt) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java]: nit: excess whitespace on this line

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java [hunk 2]
```diff
@@ -57,6 +63,6 @@ String getLogs() {
     }
 
     void clearLogs() {
-        mLogs = new StringBuffer();
+        mLogs = new StringBuilder();
```
Comment (addev-ashish) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java]: nit: set mLogSize to 0 here for completeness.

**Comment by addev-ashish** [comment]
:shipit:

**Comment by heidijinxujia** [comment]
Hi @weijjia could you squash the commits and merge the PR? We could close the issue then.

**Comment by weijjia** [comment]
nope, I'm not going to squash the commits, it's supper clear and neat, one is original changes and the other one is to address comments.

**Pull request merged**